services/oauth/v2/oauth_test.go

```diff
@@ -0,0 +1,785 @@
+			expectedResponse := &v2.AuthResponse{
+				Err:          "timeout",
+				ErrorMessage: "control plane service is having a problem: mock mock 127.0.0.1:1234->127.0.0.1:12340: read: operation timed out",
+			}
+			Expect(response).To(Equal(expectedResponse))
+			Expect(err).To(MatchError(fmt.Errorf("failed to fetch/refresh token inside getTokenInfo: %w", fmt.Errorf("error occurred while fetching/refreshing account info from CP: control plane service is having a problem: mock mock 127.0.0.1:1234->127.0.0.1:12340: read: operation timed out"))))
+		})
+	})
+
+	Describe("Test AuthStatusToggle function", func() {
+		It("authStatusToggle function call when config backend api call failed", func() {
+			ctrl := gomock.NewController(GinkgoT())
+			mockTokenProvider := mock_oauthV2.NewMockTokenProvider(ctrl)
+			mockTokenProvider.EXPECT().Identity().Return(nil)
+			mockCpConnector := mock_oauthV2.NewMockControlPlaneConnectorI(ctrl)
+			oauthHandler := &v2.OAuthHandler{
+				CacheMutex:                rudderSync.NewPartitionRWLocker(),
+				Cache:                     v2.NewCache(),
+				Logger:                    logger.NewLogger().Child("MockOAuthHandler"),
+				AuthStatusUpdateActiveMap: make(map[string]bool),
+				TokenProvider:             mockTokenProvider,
+				CpConn:                    mockCpConnector,
+			}
+			mockCpConnector.EXPECT().CpApiCall(gomock.Any()).Return(http.StatusBadRequest, `{
+				  "message":"unable to update the auth status for the destination"
+			  }`)
+			statusCode, response := oauthHandler.AuthStatusToggle(&v2.AuthStatusToggleParams{
+				Destination: &backendconfig.DestinationT{
+					ID:                    "destinationID",
+					DestinationDefinition: backendconfig.DestinationDefinitionT{Name: "testDest"},
+				},
+				WorkspaceId:     "workspaceID",
+				RudderAccountId: "rudderAccountId",
+				StatPrefix:      "AuthStatusInactive",
+				AuthStatus:      v2.AUTH_STATUS_INACTIVE,
+			})
+			Expect(statusCode).To(Equal(http.StatusBadRequest))
+			Expect(response).To(Equal("Problem with user permission or access/refresh token have been revoked"))
+		})
+		It("authStatusToggle function call when config backend api call succeeded", func() {
+			ctrl := gomock.NewController(GinkgoT())
+			mockTokenProvider := mock_oauthV2.NewMockTokenProvider(ctrl)
+			mockTokenProvider.EXPECT().Identity().Return(nil)
+			mockCpConnector := mock_oauthV2.NewMockControlPlaneConnectorI(ctrl)
+			oauthHandler := &v2.OAuthHandler{
+				CacheMutex:                rudderSync.NewPartitionRWLocker(),
+				Cache:                     v2.NewCache(),
+				Logger:                    logger.NewLogger().Child("MockOAuthHandler"),
+				AuthStatusUpdateActiveMap: make(map[string]bool),
+				TokenProvider:             mockTokenProvider,
+				CpConn:                    mockCpConnector,
+			}
+			mockCpConnector.EXPECT().CpApiCall(gomock.Any()).Return(http.StatusOK, `{
+				  "message":"unable to update the auth status for the destination"
```

Comment:
Should we update to a success message ?

Reply:
CpApiCall will be success, when authStatus has been updated successfully right ?! But you're returning an error message in the test-case. So wanted to confirm if we have to change the mock basically